### DISCUSSION TITLE
Automatically download hydat on import if version is out of date

### DISCUSSION THREAD

**Discussion by SteveCoss** [ropensci/tidyhydat#182] [Q&A] [answered]
I know how to force download, but I'd like to try and use whatever the import message is putting out to know if I have the updated version before I run download_hydat(dl_hydat_here = NULL, ask = FALSE).

**Comment by boshek** [comment, marked as answer]
Hi @SteveCoss 

Does `tidyhydat::hy_version()` do what you want?

**Reply by SteveCoss** [reply]
Hi @boshek 
That's definitely half of the problem. I also need to be able to automate checking what my local version is. I The library does this on it's own when importing, but I'm not sure how. I'm trying to include a bunch of hydat retrievals in a cluster run so having version checking automated is going to be a huge help.

**Reply by boshek** [reply]
Ok so IIUC correctly you want to be able to check the remote (i.e. the ECCC website) to see if anything what available there and then compare that to what you have locally? 

I worked a bit on a new function called `hy_remote()`. I'm still not sure what it should return but here it is in action in its more basic from. How does that look?

``` r
library(tidyhydat)

hy_version()$Date
#> [1] "2022-07-21 08:22:43 UTC"

hy_remote()
#> [1] "20220721"
```

Is this something that might be workable for you?

**Reply by SteveCoss** [reply]
I think that will be perfect! Thanks so much!
-S
